docs/gettinginvolved/plugins/building-driver.rst

```diff
@@ -0,0 +1,161 @@
+          """
+
+   If you wish the tests to cleanup volumes after each run, please provide a cleanup version of ``IBlockDeviceAPI``.
+   For an example of a clean up script, see the `EBS API with cleanup <https://github.com/ClusterHQ/flocker/blob/master/flocker/node/agents/test/blockdevicefactory.py>`_ inside ``api_factory``.
+
+   You can run these tests with the ``trial`` test runner, provided by `Twisted <(http://twistedmatrix.com/trac/wiki/TwistedTrial)>`_, one of Flocker's dependencies:
+
+   .. prompt:: bash $
+
+      trial yourstorage.test_yourstorage
+
+#. Additional functional tests:
+
+   You are encouraged to write additional functional tests to cover logic specific to your driver implementation.
+   For example, here are some `EBS driver-specific tests <https://github.com/ClusterHQ/flocker/blob/master/flocker/node/agents/functional/test_ebs.py>`_ written by ClusterHQ.
+
+#. Run acceptance tests:
+
+   After all functional tests pass, please run acceptance tests according to our :ref:`acceptance-testing` documentation.
+
+#. Setup Continuous Integration environment for tests.
+
+   After your acceptance tests run clean, please set up a CI environment for functional and acceptance tests for your driver.
+   For example:
+   
+   * `EBS functional tests <http://build.clusterhq.com/builders/flocker%2Ffunctional%2Faws%2Fubuntu-14.04%2Fstorage-driver>`_
+   * `EBS acceptance tests <http://build.clusterhq.com/builders/flocker%2Facceptance%2Faws%2Fubuntu-14.04%2Faws>`_
+
+
+Publishing Your Driver
+======================
+
+Once CI test runs pass for a week, you are ready to publish your driver and assert that is certified to work with Flocker.  
+
+Completed drivers should be published as a ``Public`` repository on GitHub.
+
+Please include the Apache 2.0 License as part of the repository.
+For example, see the `Flocker License <https://github.com/ClusterHQ/flocker/blob/master/LICENSE>`_ .
+
+
+Certifying Your Driver
+======================
+
+To demonstrate that your driver passes all tests, you should include a Build Status badge at the top of the ``README`` on your driver's GitHub repository.
+
+Examples of status images include `Travis CI <http://docs.travis-ci.com/user/status-images/>`_ and `Jenkins <https://wiki.jenkins-ci.org/display/JENKINS/Embeddable+Build+Status+Plugin>`_.
+
+You should also clearly indicate which version of Flocker your driver has been certified against.
+
+
+Enabling Flocker Users to Install Your Storage Driver
+=====================================================
+
+Once you've implemented your storage backend you'll want to allow Flocker users to use your package.
+The basic implementation strategy is that your user installs a Python package with your backend implementation on all Flocker nodes:
+
+.. prompt:: bash $
+
+    /opt/flocker/bin/pip install https://example.com/your/storageplugin-1.0.tar.gz
+
+You can also provide RPMs or DEBs that have same effect of installing a new Python package.
```

Comment:
If you think that it would be useful, create a JIRA issue to document how to do this and put a comment here linking to it.